pkg/utils/utils.go

```diff
@@ -238,17 +240,41 @@ func HasDpdkDriver(pciAddr string) (bool, error) {
 
 // SaveNetConf takes in container ID, data dir and Pod interface name as string and a json encoded struct Conf
 // and save this Conf in data dir
-func SaveNetConf(cid, dataDir, podIfName string, conf interface{}) error {
-	netConfBytes, err := json.Marshal(conf)
+func SaveNetConf(cid, dataDir, podIfName string, netConf *sriovtypes.NetConf) error {
+	netConfBytes, err := json.Marshal(netConf)
```

Comment:
Thx for giving my idea a go @SchSeba !

can we roll this logic into:

`func (nc *NetConf) MarshalJSON() ([]byte, error) {...}`

which would be part of types.go. essentially implement Marshaler interface[1]

that way, here we will call json.Marshal(conf) like we did before.


[1] https://pkg.go.dev/encoding/json#Marshaler

Reply:
yep you are right that is a better idea can you take another look please